### PR TITLE
Web Content Accessibility Guidelines improvments

### PR DESCRIPTION
**Pull request by rembiesa**

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #16126 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/16126)

File: layouts/basic/modules/Vtiger/List/Field/Base.tpl [hunk 1]
```diff
@@ -21,7 +21,7 @@
 	<div class="searchField {if isset($CLASS_SIZE)}{$CLASS_SIZE}{/if} u-min-w-150pxr">
 		{if !empty($MODULE_MODEL) && $MODULE_MODEL->getAlphabetSearchField() eq $FIELD_MODEL->getName()}
 			<div class="input-group col-12 px-0">
-				<input type="text" name="{$FIELD_MODEL->getName()}" {if !empty($FIELD_MODEL->get('source_field_name'))} data-source-field-name="{$FIELD_MODEL->get('source_field_name')}" data-module-name="{$FIELD_MODEL->getModuleName()}" {/if} class="listSearchContributor form-control" value="{$SEARCH_VALUE}" title='{$LABEL['label']}' data-fieldinfo='{$FIELD_INFO|escape}' />
+				<input type="text" name="{$FIELD_MODEL->getName()}" aria-label="{$FIELD_MODEL->getName()}" {if !empty($FIELD_MODEL->get('source_field_name'))} data-source-field-name="{$FIELD_MODEL->get('source_field_name')}" data-module-name="{$FIELD_MODEL->getModuleName()}" {/if} class="listSearchContributor form-control" value="{$SEARCH_VALUE}" title='{$LABEL['label']}' data-fieldinfo='{$FIELD_INFO|escape}' />
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/List/Field/Base.tpl]: ?

File: layouts/basic/modules/Vtiger/List/Field/Base.tpl [hunk 2]
```diff
@@ -35,7 +35,7 @@
 			</div>
 		{else}
 			<div class="input-group">
-				<input type="text" name="{$FIELD_MODEL->getName()}" {if !empty($FIELD_MODEL->get('source_field_name'))} data-source-field-name="{$FIELD_MODEL->get('source_field_name')}" data-module-name="{$FIELD_MODEL->getModuleName()}"
+				<input type="text" name="{$FIELD_MODEL->getName()}" aria-label="{$FIELD_MODEL->getName()}" {if !empty($FIELD_MODEL->get('source_field_name'))} data-source-field-name="{$FIELD_MODEL->get('source_field_name')}" data-module-name="{$FIELD_MODEL->getModuleName()}"
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/List/Field/Base.tpl]: ?

File: layouts/basic/modules/Vtiger/ListViewContents.tpl [hunk 3]
```diff
@@ -23,7 +23,7 @@
 							<button type="button" class="ml-2 btn {if !empty($ORDER_BY)}btn-info{else}btn-outline-info{/if} btn-xs js-show-modal"
 								data-url="index.php?view=SortOrderModal&module={$MODULE_NAME}"
 								data-modalid="sortOrderModal-{\App\Layout::getUniqueId()}" data-js="click">
-								<span class="fas fa-sort"></span>
+								<span aria-label="{\App\Language::translate('LBL_SORTING_SETTINGS')}" class="fas fa-sort"></span>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/ListViewContents.tpl]: ?

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

File: layouts/basic/modules/Vtiger/List/Field/Base.tpl [hunk 4]
```diff
@@ -21,7 +21,7 @@
 	<div class="searchField {if isset($CLASS_SIZE)}{$CLASS_SIZE}{/if} u-min-w-150pxr">
 		{if !empty($MODULE_MODEL) && $MODULE_MODEL->getAlphabetSearchField() eq $FIELD_MODEL->getName()}
 			<div class="input-group col-12 px-0">
-				<input type="text" name="{$FIELD_MODEL->getName()}" {if !empty($FIELD_MODEL->get('source_field_name'))} data-source-field-name="{$FIELD_MODEL->get('source_field_name')}" data-module-name="{$FIELD_MODEL->getModuleName()}" {/if} class="listSearchContributor form-control" value="{$SEARCH_VALUE}" title='{$LABEL['label']}' data-fieldinfo='{$FIELD_INFO|escape}' />
+				<input type="text" name="{$FIELD_MODEL->getName()}" role="input" {if !empty($FIELD_MODEL->get('source_field_name'))} data-source-field-name="{$FIELD_MODEL->get('source_field_name')}" data-module-name="{$FIELD_MODEL->getModuleName()}" {/if} class="listSearchContributor form-control" value="{$SEARCH_VALUE}" title='{$LABEL['label']}' data-fieldinfo='{$FIELD_INFO|escape}' />
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/List/Field/Base.tpl]: ?

File: layouts/basic/modules/Vtiger/Pagination.tpl [hunk 5]
```diff
@@ -61,19 +61,19 @@
 					{/if}
 					<li class="js-page--set page-item pageNumber{if $PAGE_NUMBER eq $PAGE_INDEX} active disabled{/if}"
 						data-id="{$PAGE_INDEX}" data-js="click">
-						<a class="page-link" href="#">{$PAGE_INDEX}</a>
+						<a class="page-link" aria-label="{\App\Language::translate('LBL_CURRENT_PAGE')}{$PAGE_INDEX}" href="#">{$PAGE_INDEX}</a>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Pagination.tpl]: ?

File: layouts/basic/modules/Vtiger/Pagination.tpl [hunk 6]
```diff
@@ -61,19 +61,19 @@
...
 					</li>
 				{/for}
 			{/if}
 			{if $PAGE_INDEX <= $PAGE_COUNT}
 				<li class="js-page--set pageNumber{if $PAGE_NUMBER eq $PAGE_COUNT} active disabled{/if}" data-js="click"
 					data-id="{$PAGE_COUNT}">
-					<a class="page-link" href="#">{$PAGE_COUNT}</a>
+					<a class="page-link" aria-label="{\App\Language::translate('LBL_CURRENT_PAGE')}{$PAGE_INDEX}" href="#">{$PAGE_COUNT}</a>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Pagination.tpl]: ?

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

File: layouts/basic/modules/Vtiger/Pagination.tpl [hunk 7]
```diff
@@ -61,14 +61,14 @@
...
-						<a class="page-link" aria-label="{\App\Language::translate('LBL_CURRENT_PAGE')}{$PAGE_INDEX}" href="#">{$PAGE_INDEX}</a>
+						<a class="page-link" {if $PAGE_INDEX === ($PAGE_COUNT - 1)}aria-label="{\App\Language::translate('LBL_PREV_PAGE')}" {else}aria-label="{\App\Language::translate('LBL_GO_TO_PAGE_NUMBER')} {$PAGE_INDEX}" {/if} href="#">{$PAGE_INDEX}</a>
 					</li>
 				{/for}
 			{/if}
 			{if $PAGE_INDEX <= $PAGE_COUNT}
 				<li class="js-page--set pageNumber{if $PAGE_NUMBER eq $PAGE_COUNT} active disabled{/if}" data-js="click"
 					data-id="{$PAGE_COUNT}">
-					<a class="page-link" aria-label="{\App\Language::translate('LBL_CURRENT_PAGE')}{$PAGE_INDEX}" href="#">{$PAGE_COUNT}</a>
+					<a class="page-link" aria-label="{\App\Language::translate('LBL_CURRENT_PAGE')} {$PAGE_COUNT}" href="#">{$PAGE_COUNT}</a>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Pagination.tpl]: ?

File: layouts/basic/modules/Vtiger/ListViewContents.tpl (hunk 3 above)
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/ListViewContents.tpl]: span? 
move to button

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

File: layouts/basic/modules/Vtiger/ListViewContents.tpl [hunk 8]
```diff
@@ -20,7 +20,7 @@
 						<label class="sr-only" for="listViewEntriesMainCheckBox">{\App\Language::translate('LBL_SELECT_ALL')}</label>
 						<input type="checkbox" id="listViewEntriesMainCheckBox" title="{\App\Language::translate('LBL_SELECT_ALL')}" />
 						{if $MODULE_MODEL->isAdvSortEnabled()}
-							<button type="button" class="ml-2 btn {if !empty($ORDER_BY)}btn-info{else}btn-outline-info{/if} btn-xs js-show-modal"
+							<button type="button" aria-label="{\App\Language::translate('LBL_SORTING_SETTINGS')}" class="ml-2 btn {if !empty($ORDER_BY)}btn-info{else}btn-outline-info{/if} btn-xs js-show-modal"
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/ListViewContents.tpl]: switch to title

**Pull request merged**